architecture/2025-04-aci-confidential-containers.md

```diff
@@ -0,0 +1,388 @@
+flow we aim to achieve through this proposal.
+
+When users interact with Radius through the CLI, include sample 
+input commands and their corresponding output. Include a bicep/helm code 
+sample, if this proposal involves updates to that experience.
+-->
+
+Ted is a developer trying to run his applications on a serverless compute platform. His application needs to run on confidential containers. His E2E experience defining a Bicep template using ACI compute looks like this: 
+
+**Setup**
+
+1. Run `rad init`.
+2. When prompted to create an environment, choose the `aci` compute option.
+3. When prompted, choose whether or not to support confidential containers. This will download the Azure CLI and the `confcom` extension. 
+
+Note: The Kubernetes flow will not change, minus the extra step to select Kubernetes compute. 
+
+**Sample Input:**
+<!--
+Provide a sample CLI command input and/or bicep/helm code.
+-->
+```
+rad init --full 
+
+Enter an environment name              
+> default  
+                                                      
+Select the compute for the environment
+  >  1. kubernetes 
+  >  2. aci
+
+Setup support for confidential containers? Selecting 'Yes' will install the Azure CLI and the `confcom` extension that are required to support confidential containers on ACI.  
+  >  1. Yes
+  >  2. No    
+```
+
+**Authoring:**
+<!--
+Provide a sample output for the inputs provided above.
+-->
+
+Ted now has a default environment setup using `ACI` as the underlying compute. He can define a Radius container that runs with a `Confidential` SKU. Ted can set additional values on the container group profile like `name`, `osType`, etc to pass through when Radius creates the container group profile. 
+
+Note: The `runtimes` property is not meant to be a replacement for defining a `ContainerGroupProfile` resource in Bicep. It is meant to be used as a punchthrough for specific values that the user wants to set on the default container group profile that Radius creates.  
+
+**Sample Bicep Template:**
+
+```diff
+resource demo 'Applications.Core/containers@2023-10-01-preview' = {
+  name: 'demo'
+  properties: {
+    application: application
+    container: {
+      image: 'ghcr.io/radius-project/samples/demo:latest'
+      ports: {
+        web: {
+          containerPort: 3000
+        }
+      }
++     // all the other container properties should also be implemented (e.g. `env`, `readinessProbe`, `livenessProbe`, etc.)
+    }
+    extensions: [
++     {
++       kind:  'manualScaling'
++       replicas: 2
++     }
+    ]
+    runtimes: {
++     aci: {
++       containergroupprofile: {
++          name: 'mycgname'
++       // Add ACI-specific properties here to punch-through the Radius abstraction, e.g. sku, osType, etc. These should be in the same format as the Bicep definition for a container group profile 
++         properties: {
++             sku: 'Confidential' // 'Standard', 'Dedicated', etc.
```

Comment:
Would this schema be exactly the same as the [current schema for container group profiles](https://learn.microsoft.com/en-us/azure/container-instances/container-instance-ngroups/container-instances-about-ngroups#container-group-profile-cg-profile)?

Will Radius validate the user input against the cgp schema?

Reply:
Yes, this is the plan. We'd keep it as close as possible to the Bicep schema (ideally the exact same) so there's less load in defining punchthrough properties